### PR TITLE
Bug 1877791: Release lock on KCM and KS termination

### PR DESCRIPTION
**Pull request by tnozicka**
KCM and KS need to gracefully terminate so the next replica can take over during a rollout. Graceful termination is important for giving up the lease, so another replica can become the leader without waiting 60s for the lease to expire.

KCM is especially important to be available as much as we can because it runs the endpoints controller which has to notice pods going down / rolling out and update service ASAP to stop sending traffic there.

Tried to patch it with minimal changes, we need to do it in upstream properly and then drop this. 
https://issues.redhat.com/browse/WRKLDS-209

/cc @deads2k @soltysh

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by tnozicka** [comment]
/hold 
the rpm serving is borked until this merges https://github.com/openshift/ci-tools/pull/1223

**Comment by tnozicka** [comment]
/hold cancel
https://github.com/openshift/ci-tools/pull/1223 merged

/retest

**Comment by tnozicka** [comment]
registry flake
/retest

**Pull request merged**
